### ISSUE TITLE
Aurora: regional scale

### ISSUE THREAD

**Issue by sergioleonluis** [microsoft/aurora#87]
Dear  @wesselb,

Using the examples available at [https://microsoft.github.io/aurora/example_hres_t0.html](url) as a
starting point, we are adapting the Aurora model inputs to a regional scale, focusing on areas
of approximately 20°x40° or smaller. So far, we have concentrated on comparing the
temperature values predicted by the global and regional versions of the model. We have also
compared these results with HRES T0 data. For example, the following figure shows the
surface temperature obtained from the global and regional Aurora and its difference for our
study region.

<img width="857" alt="Image" src="https://github.com/user-attachments/assets/8ac0ba23-71dc-4974-92d9-7be1ce444f14" />

A similar result was obtained when we compare the Aurora and HRES T0 results.

<img width="857" alt="Image" src="https://github.com/user-attachments/assets/614f216e-d770-4411-95d7-6e9206d35f8d" />

We initially expected the largest temperature differences between the global and regional
Aurora predictions to occur at the edge of the selected area, while the central region would
show very similar values in both simulations. However, the largest differences observed over
land surfaces, specifically over the northwest African coast and the Canary Islands, whereas
there is strong agreement over the oceanic region. 
Could you help us to understand why aurora predicts with so many differences the temperature values over the ocean and the land?

We understand the need for fine-tunning specific to our study region but before this task,
what other factors would you consider important as input parameters for the model? For
example, window_size, decoder_depths, etc. We believe that some of these parameters
should modify to better adapt Aurora to a regional scale, but we do not know which ones
might be more important in this respect.

Thank you in advance for your help

Regards,
Sergio León

**Comment by wesselb** [comment]
Hey @sergioleonluis! Thanks for opening an issue. :)

Very exciting that you're attempting to make regional predictions with Aurora. This is something that we're highly interested in.

As you correctly point out, the model will need to be fine-tuned for local predictions. Pretrained has been done exclusively on global data, so it's fascinating to see that the model's predictions for a local region are somewhat sensible and do not completely fall over.

> Could you help us to understand why aurora predicts with so many differences the temperature values over the ocean and the land?

Currently, this is really difficult to answer. Because the model was pretrained only on global data, the model's performance for a local region without any fine-tuning is completely up in the air. Specifically for local predictions it is important to do fine-tuning before you do any systematic study or evaluation.

> We understand the need for fine-tunning specific to our study region but before this task,
what other factors would you consider important as input parameters for the model? For
example, window_size, decoder_depths, etc. We believe that some of these parameters
should modify to better adapt Aurora to a regional scale, but we do not know which ones
might be more important in this respect.

For the initial attempt, I would keep the architecture the same, and fine-tune all parameters on the desired local region.

**Comment by wesselb** [comment]
@whx57 I would think that it is better to directly input local data into the model (at potentially a higher resolution than a crop of global data). You will, however, need to find some way to inform the model about what's happening outside of the local field of view, if that makes sense.

**Comment by gmancusi** [comment]
@wesselb during finetuning do you think it might be useful to adapt the positional embedding to inform the model of the new local area such that we can transfer the global model knowledge to the local area? If so, how do you suggest proceeding?

**Comment by wesselb** [comment]
Hey @gmancusi! When applying Aurora to local forecasting, you might need to consider two aspects: (1) The latitude and longitude spacing might become much smaller, so you'll need to double check that the current embedding of the latitude and longitude is still appropriate. It is also important to consider the numerics here. E.g., if you run things in pure bfloat16, you'll need to ensure that there is enough numerical accuracy to distinguish different lats/lons. (2) The internal Swin transformer wraps around in the longitude direction, which makes sense for global forecasting. You'll probably want to turn this off. This is currently not configurable, but should be relatively easy to adjust.
